src/labone/nodetree/node.py

```diff
@@ -0,0 +1,1038 @@
+        | NestedDict[list[list[NormalizedPathSegment]] | dict],
+        value_structure: TreeProp,
+        timestamp: int,
+        path_aliases: dict[
+            tuple[NormalizedPathSegment, ...],
+            tuple[NormalizedPathSegment, ...],
+        ]
+        | None = None,
+    ) -> None:
+        """Tree-structure representing the results of a former request.
+
+        Args:
+            path_segments: A tuple describing the path.
+            subtree_paths: Nested dictionaries, defining which sub-nodes exist.
+            value_structure: Storage of the values at the leaf-nodes. They will be
+                returned once the tree is traversed.
+            timestamp: The time the results where created.
+            path_aliases: When creating sub-nodes, these aliases are used to redirect
+                certain paths. This attribute is useful for creating artificial nodes
+                outside the normal structure. It will not be used in most other cases.
+        """
+        super().__init__(path_segments, subtree_paths, path_aliases)
+        self._value_structure: TreeProp = value_structure
+        self._timestamp = timestamp
+
+    def __getattr__(self, next_segment: str) -> ResultNode | AnnotatedValue:
+        """Go deeper into the tree structure.
+
+        Args:
+            next_segment: Segment, with which the current path should be extended
+                like this: /current/path/next_segment
+
+        Returns: New node-object, representing the extended path, or plain value,
+            if came to a leaf-node.
+
+        """
+        return self._try_generate_subnode(normalize_path_segment(next_segment))
+
+    def __getitem__(self, path_extension: str | int) -> ResultNode | AnnotatedValue:
+        """Go one or multiple levels deeper into the tree structure.
+
+        This operator can deal with a number of different scenarios, including:
+        - simple path extensions: node['deeper']
+        - path indexing: node['deeper/path']
+        - numeric indexing: node[0]
+        - wildcards (placeholder for multiple path-extensions): node['*']
+        - combinations of all that: node['deeper/*/path/0']
+
+        Args:
+            path_extension: path, number or wildcard.
+
+        Returns: New node-object, representing the extended path, or plain value,
+            if came to a leaf-node.
+
+        """
+        relative_path_segments = split_path(str(path_extension))
+        current_node = self
+        try:
+            for path_segment in relative_path_segments:
+                current_node = current_node.__getattr__(path_segment)  # type: ignore[assignment]
+
+        except AttributeError as e:
+            msg = (
+                f"Path {join_path(*self.path_segments,*relative_path_segments)} "
+                f"is invalid, because {current_node.path} "
+                f"is already a leaf-node."
+            )
+            raise RuntimeError(msg) from e
```

Comment:
You are right, this is not about the class leaf-node. Let me give you an example:
```python
result_node["config/open/too/long"]

KeyError: 'Path /zi/config/open/too/long is invalid, because /zi/config/open is already a leaf-node.'
```

Reply:
alright no I understand. But please to not use `RuntimeError` or `KeyError` but a custom, more descriptive one.